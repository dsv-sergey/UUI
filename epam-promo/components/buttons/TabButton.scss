@use  '../../assets/styles/index' as *;

@mixin tab-button-color($color, $activeColor, $backgroundColor) {
    color: $color;
    fill: $color;
    background-color: $backgroundColor;

    &:hover {
        color: $activeColor;
        fill: $activeColor;
    }

    &:global(.uui-active) {
        color: $activeColor;
        fill: $activeColor;
        border-color: $activeColor;
    }
}

$text-size-caption: 36px;
$text-size-count: 24px;

.root {
    @include button-reset();
    @include link-button-horizontal-layout(36px);
    @include tab-button-color(var(--text-color), var(--text-color-active), transparent);
    cursor: pointer;
    padding-left: 12px;
    padding-right: 12px;
    position: relative;
    bottom: -1px;
    display: flex;
    align-items: center;
    font-family: $font-sans-semibold;
    font-weight: normal;
    border-bottom: 3px solid transparent;

    &:global(.uui-disabled) {
        @include tab-button-color(var(--text-color-disabled), var(--text-color-disabled), transparent);
    }

    // &:focus {
    //     .control-focus-effect();
    // }

    &.uui-notification {
        &::after {
            content: '';
            margin-bottom: $text-size-caption / 2;
            height: 6px;
            width: 6px;
            background: $red;
            box-shadow: 0 1px 8px 0 $red;
            border-radius: 50%;
        }
    }

    :global(.uui-count) {
        @include text-size($text-size-count, 0);
        letter-spacing: 0;
        color: $gray60;
        background-color: $gray30;
        padding: 0 6px;
        margin-left: 6px;
        margin-right: 3px;
        border-radius: 9px;
    }

    :global(.uui-caption) {
        @include text-size($text-size-caption, 0);
    }

    // Sizes
    &.size-36 {
        height: 36px;
    }

    &.size-48 {
        height: 48px;
    }

    &.size-60 {
        height: 60px;
    }
}
