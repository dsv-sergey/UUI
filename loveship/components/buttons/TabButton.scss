@use '../../assets/styles/scss/index'  as *;

$text-size-caption: 36px;
$text-size-count: 24px;

@mixin tab-button-color($bg, $fg, $brd) {
    color: $fg;
    fill: $fg;

    &:hover {
        fill: $bg;
        color: $bg;
    }

    &:global(.uui-active) {
        fill: $bg;
        color: $bg;
        border-color: $brd;
    }
}

.root {
    @include button-reset();
    @include link-button-horizontal-layout();
    min-width: 36px;
    padding-left: 12px;
    padding-right: 12px;
    position: relative;
    bottom: -1px;
    display: flex;
    align-items: center;
    font-family: $font-sans-semibold;
    font-weight: normal;
    border-bottom: 2px solid transparent;

    // &:focus {
    //     .control-focus-effect();
    // }

    &.theme-light {
        @include tab-button-color($sky, $night700, $sky);

        &:global(.uui-disabled) {
            @include tab-button-color($night500, $night500, $night500);
        }
    }

    &.theme-dark {
        @include tab-button-color($sky, white, $sky);

        &:global(.uui-disabled) {
            @include tab-button-color($night600, $night600, $night600);
        }
    }

    &.uui-notification {
        &::after {
            content: '';
            margin-bottom: $text-size-caption / 2;
            height: 6px;
            width: 6px;
            background: $fire;
            box-shadow: 0 1px 8px 0 $fire;
            border-radius: 50%;
        }
    }

    :global(.uui-count) {
        @include text-size($text-size-count, 0);
        letter-spacing: 0;
        color: $night600;
        background-color: $night300;
        padding: 0 6px;
        margin-left: 6px;
        margin-right: 3px;
        border-radius: 9px;
    }

    &:global(.uui-active) {
        border-bottom: 2px solid;
    }

    :global(.uui-caption) {
        @include text-size($text-size-caption, 0);
    }

    // Sizes
    &.size-36 {
        height: 36px;
    }

    &.size-48 {
        height: 48px;
    }

    &.size-60 {
        height: 60px;
    }
}
