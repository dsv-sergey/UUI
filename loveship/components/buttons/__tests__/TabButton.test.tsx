import React from 'react';
import renderer from "react-test-renderer";
import {TabButton} from '../TabButton';
import * as acceptIcon from "../../icons/accept-12.svg";

describe('TabButton', () => {
    it('should be rendered correctly', () => {
        const tree = renderer
            .create(<TabButton />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const tree = renderer
            .create(<TabButton
                onClick={ jest.fn() }
                icon={ acceptIcon }
                isDisabled={ false }
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});