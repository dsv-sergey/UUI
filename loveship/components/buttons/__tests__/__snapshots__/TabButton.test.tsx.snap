// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabButton should be rendered correctly 1`] = `
<div
  className="container uui-button-box uui-enabled root size-48 theme-light uui-no-left-icon uui-no-right-icon"
  href={null}
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
/>
`;

exports[`TabButton should be rendered correctly with props 1`] = `
<div
  className="container uui-button-box uui-enabled -clickable root size-48 theme-light uui-has-left-icon uui-no-right-icon"
  href={null}
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <div
    className="container uui-icon uui-enabled"
  >
    <svg
      className=""
      height={24}
      viewBox="0 0 12 24"
      width={12}
    >
      <use
        xlinkHref="#accept-12.svg"
      />
    </svg>
  </div>
</div>
`;
